components/s3_bucket/main.tf

```diff
@@ -62,6 +68,26 @@ resource "aws_s3_bucket" "bucket" {
     enabled = var.versioning || var.replication_target != null
   }
 
+  dynamic "lifecycle_rule" {
+    for_each = var.temporary_paths
+
+    content {
+      id      = md5(lifecycle_rule.value)
```

Comment:
Hmm, I'd added this because (as far as I remembered) this argument was required in the past! When I went to link to that documentation, though, it says it's optional! I'm going to try omitting and see how it goes.

Reply:
I made this change in d675657 (including removing other `id` values that should no longer be necessary).

Reply:
Yay, that worked!! It automatically generated a `tf-s3-lifecycle-20210517211330135300000001` ID for this lifecycle rule when it created it. That makes these a lot easier to work with. 😌